### DISCUSSION TITLE
Remark parses `<svelte:head>` as a link

### DISCUSSION THREAD

**Discussion by babakfp** [remarkjs/remark#1308] [Q&A] [answered]
Hi 👋

These get parsed as a link:

- `<aa:0>`
- `<aa:a>`
- `<aa:->`
- etc

And these don't:

- `</aa:a>`
- `<a:a>`
- `<aa:a >`
- `< aa:a>`

Is there a reason for this? I mean, is it a valid thing to parse something that starts with `<` and ends with `>` as a link?

If it doesn't hurt the current valid behavior of Markdown, it would be great to get it parsed as html.

**Comment by ChristianMurphy** [comment, marked as answer]
Hey @babakfp!
We follow the autolink rules commonmark https://spec.commonmark.org/0.31.2/#autolinks

---

The characters `<` and `>` are used in Markdown to define autolinks, which are absolute URIs and email addresses enclosed within those characters. The specification defines the rules for what qualifies as a valid autolink.

According to the specification, a URI autolink must start with `<`, followed by an absolute URI (a scheme, colon, and then characters other than ASCII control characters, space, `<`, and `>`) and end with `>`. Similarly, an email autolink must start with `<`, followed by an email address (matching a specific regex pattern), and end with `>`.

The examples you provided that get parsed as links (`<aa:0>`, `<aa:a>`, `<aa:->`) follow this pattern of starting with `<`, containing a valid URI-like sequence (even though the `aa` is not a real scheme), and ending with `>`. Therefore, they are interpreted as autolinks according to the specification.

On the other hand, the examples that don't get parsed as links (`</aa:a>`, `<a:a>`, `<aa:a >`, `< aa:a>`) violate the rules for autolinks in some way. For instance, `</aa:a>` starts with an invalid character sequence `</`, `<a:a>` is too short to be a scheme, `<aa:a >` has a space before the closing `>`, and `< aa:a>` has a space after the opening `<`.

Regarding your question about parsing these as HTML, the specification doesn't mention parsing arbitrary strings within `<>` as HTML tags. It's specific to autolinks, which have a strict definition. Parsing arbitrary `<>` sequences as HTML tags could potentially cause conflicts with the existing Markdown syntax and might not be desirable in a Markdown context.

**Reply by babakfp** [reply]
Thank you for the awesome explanation 👍
That makes total sense.
Thanks.
